lib/std/child_process.zig

```diff
@@ -257,6 +259,55 @@ pub const ChildProcess = struct {
         }
     }
 
+    fn collectOutputWindows(child: *ChildProcess, outs: [2]*std.ArrayList(u8), max_output_bytes: usize) !void {
+        const bump_amt = 512;
+        const handles = [_]windows.HANDLE{ child.stdout.?.handle, child.stderr.?.handle };
+
+        for ([_]u1{ 0, 1 }) |i| {
+            try outs[i].ensureCapacity(bump_amt);
+            const buf = outs[i].unusedCapacitySlice();
+            child.overlapped[i] = mem.zeroes(windows.OVERLAPPED);
+            _ = windows.kernel32.ReadFile(handles[i], buf.ptr, @intCast(u32, buf.len), null, &child.overlapped[i]);
+        }
+
+        var wait_objects = handles;
+        var wait_object_count: u2 = wait_objects.len;
+
+        while (true) {
+            const status = windows.kernel32.WaitForMultipleObjects(wait_object_count, &wait_objects, 0, windows.INFINITE);
+            if (status == windows.WAIT_FAILED) {
+                switch (windows.kernel32.GetLastError()) {
+                    else => |err| return windows.unexpectedError(err),
+                }
+            }
+            if (status < windows.WAIT_OBJECT_0 or status > windows.WAIT_OBJECT_0 + wait_object_count - 1)
+                unreachable;
+
+            const wait_idx = status - windows.WAIT_OBJECT_0;
+            const i = @as(u1, if (wait_objects[wait_idx] == handles[0]) 0 else 1);
+            var read_bytes: u32 = undefined;
+            const overlapped_result = windows.kernel32.GetOverlappedResult(handles[i], &child.overlapped[i], &read_bytes, 0);
+            if (overlapped_result == 0) switch (windows.kernel32.GetLastError()) {
+                .BROKEN_PIPE => {
+                    wait_object_count -= 1;
+                    if (wait_object_count == 0)
+                        break;
+                    if (wait_idx == 0)
```

Comment:
I very much prefer my `swap` trick and my comment explaining _why_ this is done.

Reply:
I don't understand why we're doing a swap when we don't need to.  We just need to move the last element to the first element.  Why do we need to overwrite the last element if we are no longer using it?

Reply:
No reason, I liked the swap.
Don't try to pull the microoptomization card on me, that won't work.

Reply:
It's not about optimization it's about confusion.  If I see a swap, it tells me that both values are still being used, but in this case they're not.

Reply:
The swap-remove-decrement Len is a well known pattern to remove elements, let's keep the single move to avoid confusion ;)

Reply:
Interesting I'm not familiar with this pattern.  I tried googling it but I'm not finding anything (I find stuff related to sorting only).  Do you have any links or code you could point me to that uses swap to remove an item from a list?

Reply:
Check out ArrayList `swapRemove` then.

Reply:
`swapRemove` doesn't do a swap like you were doing here.
```zig
        pub fn swapRemove(self: *Self, i: usize) T {
            if (self.items.len - 1 == i) return self.pop();

            const old_item = self.items[i];
            self.items[i] = self.pop();
            return old_item;
        }

```

As you can see, it assign anything to `self.items[i-1]`.

Reply:
That's a bit different but if you squint enough you can see how the elements are swapped w/ an intermediate variable.

The same method can be implemented in terms of a single swap between the ith element and the last one, plus decrementing the length.